src/App/ComplexGeoDataPyImp.cpp

```diff
@@ -292,6 +295,231 @@ PyObject* ComplexGeoDataPy::transformGeometry(PyObject *args)
     }
 }
 
+PyObject* ComplexGeoDataPy::getElementName(PyObject* args)
+{
+    char* input;
+    int direction = 0;
+    if (!PyArg_ParseTuple(args, "s|i", &input, &direction)) {
+        return NULL;
+    }
+
+    Data::MappedElement res = getComplexGeoDataPtr()->getElementName(input);
+    std::string s;
+    if (direction == 1) {
+        return Py::new_reference_to(Py::String(res.name.appendToBuffer(s)));
+    }
+    else if (direction == 0) {
+        return Py::new_reference_to(Py::String(res.index.appendToStringBuffer(s)));
+    }
+    else if (Data::IndexedName(input)) {
+        return Py::new_reference_to(Py::String(res.name.appendToBuffer(s)));
+    }
+    else {
+        return Py::new_reference_to(Py::String(res.index.appendToStringBuffer(s)));
+    }
+}
+
+PyObject* ComplexGeoDataPy::getElementIndexedName(PyObject* args)
+{
+    char* input;
+    PyObject* returnID = Py_False;
+    if (!PyArg_ParseTuple(args, "s|O", &input, &returnID)) {
+        return NULL;
+    }
+
+    ElementIDRefs ids;
+    Data::MappedElement res =
+        getComplexGeoDataPtr()->getElementName(input, PyObject_IsTrue(returnID) ? &ids : nullptr);
+    std::string s;
+    Py::String name(res.index.appendToStringBuffer(s));
+    if (!PyObject_IsTrue(returnID)) {
+        return Py::new_reference_to(name);
+    }
+
+    Py::List list;
+    for (auto& id : ids) {
+        list.append(Py::Long(id.value()));
+    }
+    return Py::new_reference_to(Py::TupleN(name, list));
+}
+
+PyObject* ComplexGeoDataPy::getElementMappedName(PyObject* args)
+{
+    char* input;
+    PyObject* returnID = Py_False;
+    if (!PyArg_ParseTuple(args, "s|O", &input, &returnID)) {
+        return NULL;
+    }
+
+    ElementIDRefs ids;
+    Data::MappedElement res =
+        getComplexGeoDataPtr()->getElementName(input, PyObject_IsTrue(returnID) ? &ids : nullptr);
+    std::string s;
+    Py::String name(res.name.appendToBuffer(s));
+    if (!PyObject_IsTrue(returnID)) {
+        return Py::new_reference_to(name);
+    }
+
+    Py::List list;
+    for (auto& id : ids) {
+        list.append(Py::Long(id.value()));
+    }
+    return Py::new_reference_to(Py::TupleN(name, list));
+}
+
+PyObject* ComplexGeoDataPy::setElementName(PyObject* args, PyObject* kwds)
+{
+    const char* element;
+    const char* name = 0;
+    const char* postfix = 0;
+    int tag = 0;
+    PyObject* pySid = Py_None;
+    PyObject* overwrite = Py_False;
+
+    static char* kwlist[] = {"element", "name", "postfix", "overwrite", "sid", "tag", NULL};
+    if (!PyArg_ParseTupleAndKeywords(args,

```

Comment:
Should use the new wrapper method, `Wrapped_ParseTupleAndKeywords`, and then we can use std::array for kwlist.